### PR TITLE
Guide to using GeoJsonDataSource

### PR DESCRIPTION
**Pull request by jDilshodbek**
This guide was written on purpose of following [GeoJson documentation](https://github.com/maplibre/maplibre-native/issues/2527) 

Closes https://github.com/maplibre/maplibre-native/issues/2534

### CODE REVIEW

File: docs/mdbook/src/android/geojson-guide.md [hunk 1]
```diff
@@ -0,0 +1,373 @@
+# Introduction
+
+This guide will teach you how to use [`GeoJsonDataSource`]("https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.style.sources/-geo-json-source/index.html") by deep diving into [`GeoJson`]("https://geojson.org/") file format.
```
Comment (louwers) [inline docs/mdbook/src/android/geojson-guide.md:3]: Names that use own names such as GeoJSON (note capitalization) and JSON should not use backticks. That should only be done for names of classes / variables.

File: docs/mdbook/src/android/geojson-guide.md [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+
+You will start with fundamentals of how map renders data under the hood and why [`JSON`]("https://en.wikipedia.org/wiki/JSON") format is preferred in storing geospatial data
+
+# Goals
+After finishing  this documentation you should be able to
+
+1. understand how [`Style`]("https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.maps/-style/index.html?query=open%20class%20Style"), [`Layer`]("https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.style.layers/-layer/index.html?query=abstract%20class%20Layer"), and [`DataSource`]("https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.style.sources/-source/index.html?query=abstract%20class%20Source") interact with each other
+2. describe `JSON` files
+3. explore building blocks of `GeoJson` data
+4. use `GeoJson` files in constructing [`GeoJsonDataSources`]("https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.style.sources/-source/index.html?query=abstract%20class%20Source")
```
Comment (louwers) [inline docs/mdbook/src/android/geojson-guide.md:13]: Nit: should be `GeoJsonSource`s

File: docs/mdbook/src/android/geojson-guide.md [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+5. update data at runtime
+
+### 1. Styles, Layers, and DataSources
+- `Style ` defines the visual representation of the map such as colors and appearance.
+
+- `Layer` controls how data should be presented to the user
+
+- `Datasource`  holds actual data and provides layers with it
+
+Styles consist of collections of layers and datasources. Layers reference datasources. Hence they require unique source ID when they are constructed
```
Comment (louwers) [inline docs/mdbook/src/android/geojson-guide.md:23]:
```suggestion
Styles consist of collections of layers and datasources. Layers reference datasources. Hence they require a unique source ID when they are constructed
```

File: docs/mdbook/src/android/geojson-guide.md [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+Firstly, we need to understand how to store data and passing them into datasources and therefore `JSON` comes into play in the next session.
+
+### 2. JSON
+`JSON` stands for JavaScript Object Notation. It is a key-value pair text format for storing and retrieving data between client and server. 
+
+It is popular because it is human readable and easy to parse into objects.
+
+JSON content is treated as string and it makes easy to transmit data across clients. JSON file is saved in .json extension
+
+A sample json file:
```
Comment (louwers) [inline docs/mdbook/src/android/geojson-guide.md:36]: JSON is capitalized

File: docs/mdbook/src/android/geojson-guide.md [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+    "email": "johndoe@example.com",
+    "phone": "123-456-7890"
+  },
+  "hobbies": ["reading", "traveling", "coding"]
+}
+
+```
+With its straightforward syntax, we can benefit from it by describing our geospatial data inside json
+
+### 3. GeoJson
```
Comment (louwers) [inline docs/mdbook/src/android/geojson-guide.md:52]: GeoJSON

**Review by louwers — COMMENTED** [review]
General comment: this guide, while providing useful information, also has some unnatural wording. Maybe you can use tools like ChatGPT to help out. For example: "A GeoJsonDataSource might be constructed with several approaches" is grammatically correct, but "There are several ways to construct a GeoJsonDataSource" is much better. https://chatgpt.com/share/671f749c-c8e4-8008-b35e-28277e63bc38 Be careful with ChatGPT because it can also generate long unnatural flowery sentences that are not to the point.

There are also quite some grammatical issues that I cannot all address in a review. https://www.grammarly.com/ is another tool you might want to try.

Google has a good technical writing course that you might find helpful. https://developers.google.com/tech-writing They have a section on [active vs. passive voice](https://developers.google.com/tech-writing/one/active-voice). An active voice can help improve sentences like:

> Probably the most advanced use case might be animating random objects such as cars. 

You also have a lot of paragraphs that are a single sentence, which is uncommon. Again, the course has some advice on writing good paragraphs.

**Comment by jDilshodbek** [comment]
reformatted and refined grammatically and logically with ChatGPT.

File: docs/mdbook/src/android/geojson-guide.md (hunk 1 above)
Comment (louwers) [reply docs/mdbook/src/android/geojson-guide.md]: This conversation was resolved, but there are still instances where GeoJSON and other names of things that are not classes or variable names use backticks.

File: docs/mdbook/src/android/geojson-guide.md [hunk 6]
```diff
@@ -0,0 +1,349 @@
+# Introduction
+
+This guide will teach you how to use [`GeoJsonSource`]("https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.style.sources/-geo-json-source/index.html") by deep diving into [`GeoJSON`]("https://geojson.org/") file format.
+You will start with fundamentals of how a map renders data internally and why you should prefer [JSON]("https://en.wikipedia.org/wiki/JSON") format in storing geospatial data
+
+# Goals
```
Comment (louwers) [inline docs/mdbook/src/android/geojson-guide.md:6]: Please use proper spacing for Markdown. I.e. have a newline after headings, when a list starts, etc.

File: docs/mdbook/src/android/geojson-guide.md [hunk 7]
```diff
@@ -0,0 +1,349 @@
...
+After finishing  this documentation you should be able to:
+1. understand how [`Style`]("https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.maps/-style/index.html?query=open%20class%20Style"), [`Layer`]("https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.style.layers/-layer/index.html?query=abstract%20class%20Layer"), and [`Data source`]("https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.style.sources/-source/index.html?query=abstract%20class%20Source") interact with each other
+2. describe JSON files
+3. explore building blocks of `GeoJSON` data
+4. use `GeoJSON` files in constructing [`GeoJsonSource`]("https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.style.sources/-source/index.html?query=abstract%20class%20Source")s
+5. update data at runtime
```
Comment (louwers) [inline docs/mdbook/src/android/geojson-guide.md:12]: First letter of sentences should be capitalized

File: docs/mdbook/src/android/geojson-guide.md [hunk 8]
```diff
@@ -0,0 +1,349 @@
...
+
+### 1. Styles, Layers, and Data source
+- `Style ` defines the visual representation of the map such as colors and appearance.
```
Comment (louwers) [inline docs/mdbook/src/android/geojson-guide.md:15]: Inconsistent use of dot and no dot at the end of a sentence is a list.

File: docs/mdbook/src/android/geojson-guide.md [hunk 9]
```diff
@@ -0,0 +1,349 @@
...
+- `Layer` controls how data should be presented to the user
+- `Data source`  holds actual data and provides layers with it
+
+Styles consist of collections of layers and data source. Layers reference data source. Hence they require a unique source ID when you construct them.
+It would be meaningless if we don't have any data to show, so we need know how to supply data through a data source.
+Firstly, we need to understand how to store data and pass it into a data source; therefore, we will discuss JSON in the next session.
+
+### 2. JSON
```
Comment (louwers) [inline docs/mdbook/src/android/geojson-guide.md:23]: I think we can assume that everyone knows JSON...

We can probably link to an explanation of GeoJSON elsewhere and focus on concepts that are specific to MapLibre Android.

File: docs/mdbook/src/android/geojson-guide.md [hunk 10]
```diff
@@ -0,0 +1,349 @@
...
+style.addLayer(lineLayer)
+```
+
+Note that you can not simply show data on a map. Layers must reference them.
+Therefore, you create a layer that gives visual appearance to it.
+
+- Loading from local files
+
+with assets folder file
+```kotlin
```
Comment (louwers) [inline docs/mdbook/src/android/geojson-guide.md:149]: All code snippets should be referenced so that we can be sure they compile in the future.

**Review by louwers — COMMENTED** [review]
Better, but there are still some issues.

Also I would like to get #2978 merged before accepting new documentation content, as everything we merge now needs to be ported over to the new site.

**Pull request merged**